clients/drcachesim/tracer/instru_offline.cpp

```diff
@@ -403,17 +403,23 @@ int
 offline_instru_t::append_unit_header(byte *buf_ptr, thread_id_t tid, ptr_int_t window)
 {
     byte *new_buf = buf_ptr;
-    offline_entry_t *entry = (offline_entry_t *)new_buf;
-    entry->timestamp.type = OFFLINE_TYPE_TIMESTAMP;
-    uint64 frozen = frozen_timestamp_.load(std::memory_order_acquire);
-    entry->timestamp.usec = frozen != 0 ? frozen : instru_t::get_timestamp();
-    new_buf += sizeof(*entry);
+    new_buf += append_timestamp(new_buf);
```

Comment:
It'd be good to verify what effect this change has on trace quality metrics that rely on the timestamps, so that we can adjust thresholds if needed.

Reply:
Logically it should not affect anything as our metrics divide total time into 1000 steps and that hasn't changed.  That analyzer currently makes a schedule entry per timestamp, so it will now have more entries, with each one being smaller, but other than using more memory that should not affect its analyses: it will just make the context switch-related ones more accurate with more data points.  I did test this on a real internal trace and got a 0.96 metric and looked at the 1000 timestamp output and it all looks good, just has more thread "segments" (one per timestamp).